Sources/Plasma/PubUtilLib/plMath/plTriUtils.cpp

```diff
@@ -45,17 +45,18 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #include <cmath>
 #include "plTriUtils.h"
 
-static const float kAlmostZero = 1.e-5f;
-static const float kPastZero = -kAlmostZero;
-static const float kPastOne = 1.f + kAlmostZero;
-static const float kAlmostOne = 1.f - kAlmostZero;
-static const float kAlmostZeroSquared = kAlmostZero*kAlmostZero;
+static constexpr float kAlmostZero = 1.e-5f;
+static constexpr float kPastZero = -kAlmostZero;
+static constexpr float kPastOne = 1.f + kAlmostZero;
+static constexpr float kAlmostOne = 1.f - kAlmostZero;
+static constexpr float kAlmostZeroSquared = kAlmostZero*kAlmostZero;
```

Comment:
I don't think changing a const to a constexpr will have any change on performance. Evaluating at compile time might actually slightly increase memory use because the value might be encoded into the binary multiple times.

Reply:
constexpr in this case only guarantees that the value is computed at compile time instead of runtime. And since its declared static it will only be compiled once.

Reply:
constexpr doesn't actually guarantee that it will be evaluated at compile time. Like inline, it's a hint that the compiler may ignore.

Here's it's just a little strange because all these numbers are constants. There's nothing that could be done at runtime.  `1.f + kAlmostZero` is a constant value - not something that gets done at runtime.

Reply:
Sorry for the extra comment - was thinking about it this morning a bit more...

Maybe a better way to phrase my question: In since there is a performance claim here - was the previous way identity matrices and numerical constants were defined not compile time constant? The reason the performance is important here is because if there is an issue in the way we're defining our matrices and numerical constants - we need to fix that across the engine.

If constexpr means the same thing as const in this context - that doesn't mean I'd ask for it to be changed back. But I want to make sure we understand what is going on here for future work and PRs.

Reply:
i will admit this is more a semantic change then a functional one as far as goldbolt is concerned.

Reply:
My understanding of constexpr is it usually only adds behavior to constexpr functions. The changes in the iterator are actually a really good example of constexpr functions.

In C - unless the compiler is just very strange - numeric constants and C arrays should evaluate at compile time. C arrays are just buffers so they'll be stored in the binary and loaded into memory with the binary. Numerics should be computed at compile time. Something like 'const float halfPi = M_PI/2.f` will evaluate at compile time.

Just be aware if you're looking for performance that there is a set of rules about whats precompiled that predates constexpr. Constexpr implies const - so it hopefully shouldn't hurt anything here or in the identity matrices. But under typical C compiler rules this should all be evaluated at compile time.

Reply:
Changing these to `static constexpr` is ideal.